pyVmomi/SoapAdapter.py

```diff
@@ -1321,14 +1345,20 @@ def InvokeMethod(self, mo, info, args, outerStub=None):
       req = self.SerializeRequest(mo, info, args)
       for modifier in self.requestModifierList:
          req = modifier(req)
-      conn = self.GetConnection()
-      try:
-         conn.request('POST', self.path, req, headers)
-         resp = conn.getresponse()
-      except (socket.error, http_client.HTTPException):
-         # The server is probably sick, drop all of the cached connections.
-         self.DropConnections()
-         raise
+
+      max_retries = 100
+      num_retries = 0
+      retry = True
+      fail_next = False
+      while retry is True:
+         if num_retries == max_retries:
+            fail_next = True
+         conn, resp = self._connection(req, headers, fail_next)
+         if conn is not None:
+            retry = False
+         time.sleep(0.1)
```

Comment:
Can you also comment on how the number of retries and the sleep period between retry are selected? I am not sure under what situation the server may return "", but when it happens, does the retry really work? How many times it usually takes until it succeeds?